cmd/dep/init.go

```diff
@@ -239,6 +242,45 @@ func hasImportPathPrefix(s, prefix string) bool {
 	return strings.HasPrefix(s, prefix+"/")
 }
 
+// itemizedFeedback itemizes resolved dependencies and logs the solution as
+// feedback.
+func itemizedFeedback(m *dep.Manifest, l *dep.Lock, pkgT pkgtree.PackageTree) {
+	// Algo:
+	// loop through lock projects
+	// 1. check in manifest for for same entry
+	// 2. if exists in manifest, it's direct dep, goes to manifest & lock
+	// 3. if not exists in manifest, check in pkgT
+	//	3.a. if exists in pkgT, it's direct dep, hint constraint, lock only
+	//	3.b. if not exists in pkgT, it's transitive dep, lock only
+	// 4. log feedback
```

Comment:
Imports are the only thing responsible for deciding what's direct vs. transitive - not the manifest. (Well, `required` packages in the manifest have the same effect, but it's not possible for there to be any here, so that's a wash). So, I _think_ you can just drop 1 & 2 and go right to 3.

3.a. isn't quite on the mark either, but I think that what I wrote in the spec doc might be clearer with 1 & 2 out of the way? if not, i can clarify further 😄